risc0/zkvm/src/host/api/mod.rs

```diff
@@ -319,28 +289,45 @@ pub enum AssetRequest {
     Path(PathBuf),
 }
 
+/// Provides information about the result of execution.
+pub struct SessionInfo {
+    /// The number of segments.
+    pub segments: u32,
+
+    /// The data publicly committed by the guest program.
+    pub journal: Bytes,
```

Comment:
Maybe this should be `Option<Bytes>` since an exit code other than `Paused` or `Halted` will not have a journal. Empty bytes also seems reasonable.

Reply:
There was a Slack discussion about how "empty journal" hashes to `SHA256([])` whereas "no journal" "hashes" to `0`. If we want to retain this distinction we probably need a way to represent it here (which `Option<Bytes>` makes sense for that to me).

Reply:
Actually the hashing distinction was only really relevant to the way the circuit generates the 'output' digest.

I guess `Option<Bytes>` is slightly better for the API, but we should make that change in all the other APIs as well (this is just the client/server API, which is kind of a private API).

Reply:
I tried making this change but seems to be non-trivial so I'm going to kick this to a future PR.